src/games/strategy/triplea/attachments/TechAbilityAttachment.java

```diff
@@ -101,8 +101,11 @@ public static TechAbilityAttachment get(final TechAdvance type, final String nam
   private IntegerMap<UnitType> m_defenseRollsBonus = new IntegerMap<>();
   private IntegerMap<UnitType> m_bombingBonus = new IntegerMap<>();
 
+  private final String mapName;
```

Comment:
Is this field needed? Looks like some of the exception calls pass data and others pass mapName?